File: Software Technologies/Java Basics - Exercises/src/Phonebook.java
import java.util.HashMap;
import java.util.Scanner;

public class Phonebook {
    public static void main(String [] args) {
        Scanner scanner = new Scanner(System.in);
        String inputLine = scanner.nextLine();

        HashMap<String,String> phonebook = new HashMap<String,String>();

        while (!inputLine.equals("END")) {
            String[] comandArgs = inputLine.split("\\s");

            if (comandArgs[0].equals("A")) {
                String name = comandArgs[1];
                String phoneNumber = comandArgs[2];

                if (!phonebook.containsKey(name)) {
                    phonebook.put(name, "");
                }

                phonebook.put(name, phoneNumber);

            } else {
                String name = comandArgs[1];

                if (phonebook.containsKey(name)) {
                    System.out.println(name+" -> "+phonebook.get(name));
                } else {
                    System.out.println("Contact "+name+" does not exist.");
                }
            }

            inputLine = scanner.nextLine();
        }
    }
}

File: Programming Fundamentals/Homeworks and Labs/DictionariesLambdaAndLINQ-Exercises/01.Phonebook/Phonebook.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Phonebook
{
    public static void Main()
    {
        string inputLine = Console.ReadLine();

        var phonebook = new Dictionary<string, string>();

        while (!inputLine.Equals("END"))
        {
            string[] comandArgs = inputLine.Split().ToArray();

            if (comandArgs[0].Equals("A"))
            {
                string name = comandArgs[1];
                string phoneNumber = comandArgs[2];

                if (!phonebook.ContainsKey(name))
                {
                    phonebook.Add(name, string.Empty);
                }

                phonebook[name] = phoneNumber;
            }
            else
            {
                string name = comandArgs[1];

                if (phonebook.ContainsKey(name))
                {
                    Console.WriteLine($"{name} -> {phonebook[name]}");
                }
                else
                {
                    Console.WriteLine($"Contact {name} does not exist.");
                }
            }

            inputLine = Console.ReadLine();
        }
    }
}